hack/setup-go.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# script to setup go version with gimme as needed
+# MUST BE RUN FROM THE REPO ROOT DIRECTORY
+
+# read go-version file unless EKSD_GO_IMAGE_TAG & GO_VERSION are set
+GO_VERSION="${GO_VERSION:-"$(cat .go-version)"}"
+EKSD_GO_IMAGE_TAG="${EKSD_GO_IMAGE_TAG:-"${GO_VERSION}"}"
+GO_IMAGE=public.ecr.aws/eks-distro-build-tooling/golang:$EKSD_GO_IMAGE_TAG-gcc
+
+# we don't actually care where the .env files are
+# however, GIMME_SILENT_ENV doesn't trigger re-generating a .env if it
+# already exists and isn't "silent" (no `go version` command in it)
+# so we fix that by changing where the .env is written, ensuring ours
+# is generated from this repo and silent.
+export GIMME_ENV_PREFIX=./bin/.gimme/
+export GIMME_SILENT_ENV=y
+
+# only setup go if we haven't set FORCE_HOST_GO, or `go version` doesn't match
+# go version output looks like:
+# go version go1.14.5 darwin/amd64
+if ! ([ -n "${FORCE_HOST_GO:-}" ] || \
+      (command -v go >/dev/null && [ "$(go version | cut -d' ' -f3)" = "go${GO_VERSION}" ])); then
+    # eval because the output of this is shell to set PATH etc.
+    eval "$(hack/third_party/gimme/gimme "${GO_VERSION}")"
+fi
+
+# force go modules
+export GO111MODULE=on
```

Comment:
Isn't Go modules enabled by default?

Reply:
This is to make sure we force it on, in case it is disabled, and this `setup-go.sh` was taken from `kubernetes-sigs/kind`, which was later adopted in `kubernetes` repository as part of the work to keep Kubernetes on supported Go versions.

The gimme script has the ability to install temporarily in local environment with the exact specified Go version without interruption local Go installation. With this ability installed in a repo, when we are running any unit tests or integration tests, we can make sure we are running with the exact specified Go version from `.go-version` file.

See kubernetes adoption on gimme in: https://github.com/kubernetes/kubernetes/pull/115377

See the initial adoption of gimme in kind in: https://github.com/kubernetes-sigs/kind/blob/main/hack/build/setup-go.sh